### PR TITLE
feat: Add chart for Prometheus and Grafana integration

### PR DESCRIPTION
**Pull request by levaitamas**
Details at https://github.com/l7mp/stunner/pull/51.
This PR also adds new value to STUNner helm chart to enable monitoring. 
For example: `helm install stunner stunner/stunner --create-namespace --namespace=stunner --set stunner.deployment.monitoring.enabled=true`

Comments are welcome.

### CODE REVIEW

File: helm/stunner-prometheus/templates/prometheus-operator-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    metadata:
+      labels:
+        app.kubernetes.io/component: {{ .Values.selectorLabels.component }}
+        app.kubernetes.io/name: {{ .Values.selectorLabels.name }}
+        app.kubernetes.io/version: {{ .Values.selectorLabels.version }}
+    spec:
+      containers:
+      - args:
+        - --kubelet-service=kube-system/kubelet
+        - --prometheus-config-reloader=quay.io/prometheus-operator/prometheus-config-reloader:v0.58.0
```
Comment (davidkornel) [inline helm/stunner-prometheus/templates/prometheus-operator-deployment.yaml:27]: Maybe set the images parameterizable in values.yaml that start with quay.io?

File: helm/stunner-prometheus/templates/kube-state-metrics.yaml [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        - --telemetry-host=127.0.0.1
+        - --telemetry-port=8082
+        image: registry.k8s.io/kube-state-metrics/kube-state-metrics:{{ .Values.version.kubestatemetrics }}
+        name: kube-state-metrics
+      - args:
+        - --logtostderr
+        - --secure-listen-address=:8443
+        - --tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305
+        - --upstream=http://127.0.0.1:8081/
+        image: quay.io/brancz/kube-rbac-proxy:v0.13.0
```
Comment (davidkornel) [inline helm/stunner-prometheus/templates/kube-state-metrics.yaml:161]: Configurable?

File: helm/stunner-prometheus/templates/kube-state-metrics.yaml [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+        - containerPort: 8443
+          name: https-main
+        securityContext:
+          runAsUser: 65534
+      - args:
+        - --logtostderr
+        - --secure-listen-address=:9443
+        - --tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305
+        - --upstream=http://127.0.0.1:8082/
+        image: quay.io/brancz/kube-rbac-proxy:v0.13.0
```
Comment (davidkornel) [inline helm/stunner-prometheus/templates/kube-state-metrics.yaml:173]: Configurable? Interestingly it runs two containers with the same image :D

File: helm/stunner-prometheus/templates/prometheus-rule.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  #few alert rules example
+  - name: ./example.rules
+    rules:
+    #this one fires immediately
+    - alert: ExampleAlert
+      expr: vector(1)
+  - name: ./ctrlr_list.rules
+    rules:
+    #if more then five requests came trough, fires
+    - alert: listener-over-five-requests
```
Comment (davidkornel) [inline helm/stunner-prometheus/templates/prometheus-rule.yaml:21]: Remove this part, please. If I'm correct it was copied from the l7mp-prometheus chart. Also, I don't know if we even need a PrometheusRule or not.

File: helm/stunner-prometheus/templates/node-exporter.yaml [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        - --logtostderr
+        - --secure-listen-address=[$(IP)]:9100
+        - --tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305
+        - --upstream=http://127.0.0.1:9100/
+        env:
+        - name: IP
+          valueFrom:
+            fieldRef:
+              fieldPath: status.podIP
+        image: quay.io/brancz/kube-rbac-proxy:v0.13.0
```
Comment (davidkornel) [inline helm/stunner-prometheus/templates/node-exporter.yaml:92]: Configurable?

File: helm/stunner-prometheus/templates/prometheus-rule.yaml (hunk 4 above)
Comment (levaitamas) [reply helm/stunner-prometheus/templates/prometheus-rule.yaml]: Thanks! Fixed in [74d9e1d](https://github.com/l7mp/stunner-helm/pull/5/commits/74d9e1ddd72ff1f59c52c9f49d25c7f1b28932c5).

**Comment by levaitamas** [comment]
Thanks for the comments @davidkornel ! 

Recent changes:
- Remove the unused prometheus-rule example
- Update components to the latest stable version
- Introduce `.Values.version` and `.Values.image` to configure images and versions: https://github.com/l7mp/stunner-helm/pull/5/commits/1380885833ffeef8385bc01aa659279d0496d51e#diff-734c5b5772b0edf834f4880430d84d2fb2a294b0a7057ac502a0494d66c6e54aR41 wdyt?

_(Tested on minikube v.1.28.0 (k8s v1.25), and helm v3.10.3.)_

**Review by davidkornel — APPROVED** [review]
Thank you for your effort in updating the CRDs and making all the images parameterizable. 
LGTM!

**Pull request merged**